### PR TITLE
Record CPU and walltime from progress monitor in Verifier

### PR DESCRIPTION
**Pull request by suyucs**
Record CPU and walltime from progress monitor in Verifier. This way, we are able to know walltime and cpu time for both select and non-select queries

### CODE REVIEW

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java [hunk 1]
```diff
@@ -435,6 +436,26 @@ private QueryResult executeQuery(String url, String username, String password, Q
         }
     }
 
+    private static class RecordingProgressMonitor
```
Comment (haozhun) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: I would name it `LastQueryStatsProgressMonitor`

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java [hunk 2]
```diff
@@ -435,6 +436,26 @@ private QueryResult executeQuery(String url, String username, String password, Q
...
+        private boolean finished = false;
+
+        @Override
+        public synchronized void accept(QueryStats queryStats)
+        {
+            checkState(!finished);
+            this.queryStats = queryStats;
+        }
+
+        public synchronized QueryStats finish()
```
Comment (haozhun) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: I would name it `getLastQueryStats`.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java (hunk 1 above)
Comment (haozhun) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: We generally put nested classes at the end.

**Comment by haozhun** [comment]
Assign back to me after you address my comments.

**Comment by suyucs** [comment]
Address the reviews

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java [hunk 3]
```diff
@@ -669,4 +670,24 @@ public int compareTo(ChangedRow that)
                     .result();
         }
     }
+
+    private static class LastQueryStatsProgressMonitor
+            implements Consumer<QueryStats>
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: How about ProgressMonitor

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java [hunk 4]
```diff
@@ -669,4 +670,24 @@ public int compareTo(ChangedRow that)
...
+        private boolean finished = false;
+
+        @Override
+        public synchronized void accept(QueryStats queryStats)
+        {
+            checkState(!finished);
+            this.queryStats = queryStats;
+        }
+
+        public synchronized QueryStats getLastQueryStats()
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: Why not "getStats"?

Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: Or use "final" rather than "last".

**Comment by suyucs** [comment]
Addressed @electrum  comments

**Comment by haozhun** [comment]
Looks good

**Pull request merged**